src/cms/static/js/pages/page_bulk_actions.js

```diff
@@ -2,44 +2,58 @@
+        if (document.getElementsByName("select_all_pages")[0].checked == true) {
+            selected_pages.forEach(e => e.checked = true);
+        } else {
+            selected_pages.forEach(e => e.checked = false);
+        }
     }
-}
 
-function bulk_action_execute() {
-    var action = document.getElementById('bulk_action').value;
-    // match action types
-    if ( action == "archive_pages" ) {
-        // TODO
-    } else if (action == "pdf_export") {
-        var selected_pages = document.getElementsByName("page_selected");
-        pages = [];
-        selected_pages.forEach(e => {
-            if(e.checked) {
-                pages.push(e.value)
-            }
-        })
-        if (pages.length != 0) {
-            var pdf_export_url = document.getElementById("pdf_export_url").value;
-            window.open(pdf_export_url+"?pages="+pages.join(','));
-        } 
-    } else { // no previous match, than language code -> XLIFF export
-        var page_selected = document.getElementsByName("page_selected");
+    u("#bulk_action_btn").on('click', bulk_action_execute);
+    function bulk_action_execute() {
+        var action = u('#bulk_action').first().value;
         var pages = [];
-        page_selected.forEach(async function (page_checkbox) {
-            if(page_checkbox.checked) {
-                pages = pages.concat(page_checkbox.value);
-            }
-        });
+        u('input').filter(':checked').each((node, i) => pages.push(node.value));
         if (pages.length == 0) {
+            // abort if no pages selected
             return;
         }
-        var bulk_action_url = document.getElementById("bulk_action_url").value;
-        window.open(bulk_action_url+"?target_lang="+action+"&pages="+pages.join(','), "_blank");
+        // match action types
+        if ( action == "archive_pages" ) {
+            // TODO
+        } else if (action == "pdf_export") {
+            var pdf_export_url = document.getElementById("pdf_export_url").value;
+            window.open(pdf_export_url+"?pages="+pages.join(','));
+        } else { // no previous match, than language code -> XLIFF export
+            const state = true;
+            const url = u("#ajax_url").first().dataset.url;
+            var bulk_action_url = document.getElementById("bulk_action_url").value;
+            // download XLIFF
+            window.open(bulk_action_url+"?target_lang="+action+"&pages="+pages.join(','), "_blank");
+            updateTranslationIcon(pages, action);
+        }
+    }
+
+    function updateTranslationIcon(pages, languageCode) {
+        for (page of pages){
+            let pageNode = u(`#page-${page}`);
+            // get language for updating translation state icon
+            let targetLanguage = pageNode.find(`.${languageCode}`);
+            if (!targetLanguage.find("#translation-icon").find("span").hasClass("no-trans")) {
+                // if the page already contains a translation for the exported language
```

Comment:
Second server request is no replaced by finding the existing translations of selected pages on DOM-side.

Reply:
Awesome!